WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -0,0 +1,617 @@
+        
+        return noResultsTitles(draftMessage, scheduled: scheduledMessage, trashed: trashedMessage, published: publishedMessage)
+    }
+    
+    private func noResultsTitlesWhenFiltering() -> [PostListStatusFilter:String] {
+        
+        let draftMessage = NSLocalizedString("You don't have any drafts.", comment: "Displayed when the user views drafts in the pages list and there are no pages")
+        let scheduledMessage = NSLocalizedString("You don't have any scheduled pages.", comment: "Displayed when the user views scheduled pages in the pages list and there are no pages")
+        let trashedMessage = NSLocalizedString("You don't have any pages in your trash folder.", comment: "Displayed when the user views trashed in the pages list and there are no pages")
+        let publishedMessage = NSLocalizedString("You haven't published any pages yet.", comment: "Displayed when the user views published pages in the pages list and there are no pages")
+        
+        return noResultsTitles(draftMessage, scheduled: scheduledMessage, trashed: trashedMessage, published: publishedMessage)
+    }
+    
+    private func noResultsTitles(draft: String, scheduled: String, trashed: String, published: String) -> [PostListStatusFilter:String] {
+        return [.Draft: draft,
+                .Scheduled: scheduled,
+                .Trashed: trashed,
+                .Published: published]
+    }
+    
+    func noResultsMessageText() -> String {
+        if syncHelper?.isSyncing == true || isSearching() {
+            return ""
+        }
+        
+        let filter = currentPostListFilter()
+        
+        // currentPostListFilter() may return `nil` at this time (ie: it's been declared as
+        // `nullable`).  This will probably change once we can migrate
+        // AbstractPostListViewController to Swift, but for the time being we're defining a default
+        // filter here.
+        //
+        // Diego Rey Mendez - 2016/04/18
+        //
+        let filterType = filter?.filterType ?? .Draft
+        var message : String
+        
+        switch filterType {
+        case .Draft:
+            message = NSLocalizedString("Would you like to create one?", comment: "Displayed when the user views drafts in the pages list and there are no pages")
+            break
+        case .Scheduled:
+            message = NSLocalizedString("Would you like to schedule a draft to publish?", comment: "Displayed when the user views scheduled pages in the oages list and there are no pages")
+            break
+        case .Trashed:
+            message = NSLocalizedString("Everything you write is solid gold.", comment: "Displayed when the user views trashed pages in the pages list and there are no pages")
+            break
+        default:
+            message = NSLocalizedString("Would you like to publish your first page?", comment: "Displayed when the user views published pages in the pages list and there are no pages")
+            break
```

Comment:
This is both very interesting, and scary.  I like it though, thanks. :)

Scary because it's hard to read `switch` statements without them falling through by default!